tabrepo/repository/ensemble_mixin.py

```diff
@@ -1,83 +1,286 @@
-        :param rank: whether to return ranks or raw scores (e.g. RMSE). Ranks are computed over all base models and
-        automl framework.
-        :param folds: list of folds that need to be evaluated, use all folds if not provided.
-        :param backend: Options include ["native", "ray"].
-        :return: Tuple:
-            Pandas Series of ensemble test errors per task, with multi-index (dataset, fold).
-            Pandas DataFrame of ensemble weights per task, with multi-index (dataset, fold). Columns are the names of each config.
+        Evaluates an ensemble of a list of configs on a given task (dataset, fold).
+
+        Parameters
+        ----------
+        dataset: str
+            The dataset to evaluate
+        fold: int
+            The fold of the dataset to evaluate
+        configs: list[str], default = None
+            The list of configs to consider for ensembling.
+            If None, will use all configs.
+            Models will be simulated as being fit in the order specified in `fit_order`.
+        time_limit: float, default = None
+            The time limit of the ensemble.
+            Will only consider the first N models in `configs` whose cumulative time limit is less than `time_limit`.
+        ensemble_cls: Type[EnsembleScorer], default = EnsembleScorerMaxModels
+            The ensemble method to use.
+        ensemble_kwargs: dict, default = None
+            The ensemble method kwargs.
+        ensemble_size: int, default = 100
+            The number of ensemble iterations.
+        rank: bool, default = True
+            If True, additionally calculates the rank of the ensemble result.
+        fit_order: Literal["original", "random"], default = "original"
+            Whether to simulate the models being fit in their original order sequentially or randomly.
+        seed: int, default = 0
+            The random seed used to shuffle `configs` if `fit_order="random"`.
+
+        Returns
+        -------
+        result: pd.DataFrame
+            A single-row multi-index (dataset, fold) DataFrame with the following columns:
+                metric_error: float
+                    The ensemble's metric test error.
+                metric: str
+                    The target evaluation metric.
+                time_train_s: float
+                    The training time of the ensemble in seconds (the sum of all considered models' time_train_s)
+                time_infer_s: float
+                    The inference time of the ensemble in seconds (the sum of all non-zero weight models' time_infer_s)
+                problem_type: str
+                    The problem type of the task.
+                metric_error_val: float
+                    The ensemble's metric validation error.
+        ensemble_weights: pd.DataFrame
+            A single-row multi-index (dataset, fold) DataFrame with column names equal to `configs`.
```

Comment:
Same comment here, we should probably not return single row.

Reply:
ditto to above